src/Test/Utilities/Portable/TargetFrameworkUtil.cs

```diff
@@ -157,25 +158,64 @@ IEnumerable<string> getNames(IEnumerable<MetadataReference> e)
                 }
             }
 
-            string getName(MetadataReference m)
+            string getName(MetadataReference m) => GetAssemblyIdentity(m)?.Name;
+        }
+
+        /// <summary>
+        /// Verification only succeeds when the cor library is named exactly mscorlib. This function will calculate the 
+        /// best default for <see cref="Verification"/> based on the framework / references that we are targeting. If there
+        /// is no mscorlib then the assumption is that verification will fail.
+        /// </summary>
+        public static Verification GetVerification(TargetFramework tf, IEnumerable<MetadataReference> additionalReferences)
+        {
+            switch (tf)
+            {
+                case TargetFramework.Empty:
+                    return additionalReferences.Any(x => GetAssemblyIdentity(x)?.Name == "mscorlib")
+                        ? Verification.Passes
+                        : Verification.Fails;
+                case TargetFramework.Mscorlib40:
+                case TargetFramework.Mscorlib40Extended:
+                case TargetFramework.Mscorlib40AndSystemCore:
+                case TargetFramework.Mscorlib40AndVBRuntime:
+                case TargetFramework.Mscorlib45:
+                case TargetFramework.Mscorlib45Extended:
+                case TargetFramework.Mscorlib45AndCSharp:
+                case TargetFramework.Mscorlib45AndVBRuntime:
+                case TargetFramework.Mscorlib46:
+                case TargetFramework.Mscorlib46Extended:
+                case TargetFramework.Mscorlib461:
+                case TargetFramework.Mscorlib461Extended:
+                case TargetFramework.NetStandard20:
+                case TargetFramework.WinRT:
+                case TargetFramework.DefaultVb:
+                    return Verification.Passes;
+                case TargetFramework.Standard:
+                case TargetFramework.StandardAndCSharp:
+                case TargetFramework.StandardAndVBRuntime:
+                case TargetFramework.StandardCompat:
+                    return Verification.Fails;
```

Comment:
nit: It's probably good to leave more comments. I wouldn't know what to expect for different frameworks here, or why.